server/src/main/java/org/elasticsearch/reservedstate/service/ReservedProjectStateUpdateTask.java

```diff
@@ -78,8 +78,11 @@ protected ClusterState execute(ClusterState currentState) {
         );
         ProjectMetadata updatedProjectMetadata = updatedClusterState.getMetadata().getProject(projectId);
         return ClusterState.builder(currentState)
-            .putCustom(ProjectStateRegistry.TYPE, ProjectStateRegistry.builder(updatedProjectStateRegistry).build())
-            .putProjectMetadata(ProjectMetadata.builder(updatedProjectMetadata).put(result.v2()))
+            .putCustom(
+                ProjectStateRegistry.TYPE,
+                ProjectStateRegistry.builder(updatedProjectStateRegistry).putReservedStateMetadata(projectId, result.v2()).build()
+            )
+            .putProjectMetadata(ProjectMetadata.builder(updatedProjectMetadata))
```

Comment:
Nit: this can be just
```suggestion
            .putProjectMetadata(updatedProjectMetadata)
```